src/lib/common/sem.cpp

```diff
@@ -459,13 +459,131 @@ int timeStatSemGive(const char* who, const char* what)
 
 
 /* ****************************************************************************
+*
+* reqSemGet - 
+*/
+const char* reqSemGet(void)
+{
+  int value;
+
+  if (sem_getvalue(&reqSem, &value) == -1)
+  {
+    return "error - state not available";
+  }
+
+  if (value == 0)
+  {
+    return "taken";
+  }
+  
+  return "free";  
+}
+
+
+
+/* ****************************************************************************
+*
+* transSemGet - 
+*/
+const char* transSemGet(void)
+{
+  int value;
+
+  if (sem_getvalue(&transSem, &value) == -1)
+  {
+    return "error - state not available";
+  }
+
+  if (value == 0)
+  {
+    return "taken";
+  }
+  
+  return "free";  
+}
+
+
+
+/* ****************************************************************************
+*
+* cacheSemGet - 
+*/
+const char* cacheSemGet(void)
+{
+  int value;
+
+  if (sem_getvalue(&cacheSem, &value) == -1)
+  {
+    return "error - state not available";
+  }
+
+  if (value == 0)
+  {
+    return "taken";
+  }
+  
+  return "free";  
+}
+
+
+
+/* ****************************************************************************
+*
+* timeStatSemGet - 
+*/
+const char* timeStatSemGet(void)
+{
+  int value;
+
+  if (sem_getvalue(&timeStatSem, &value) == -1)
+  {
+    return "error - state not available";
+  }
+
+  if (value == 0)
+  {
+    return "taken";
+  }
+  
+  return "free";  
+}
+
+
+
+/* ****************************************************************************
 *  curl context
 */
 
+static bool contexts_mutex_taken   = false;
+static int  contexts_mutex_errors  = 0;
+static bool contexts_mutex2_taken  = false;
```

Comment:
'mutex2' is not a very fortunate name... I'd try to be more explicit here. Maybe mutextOuter/mutexInner?

(Actually, I don't have any idea of the differences... that's just an example to show what I mean :)